### PR TITLE
[curves] introduce wrappers for Fp and Fq

### PR DESCRIPTION
**Pull request by mimoo**
This introduces wrapper types for `ark_ff::BigInteger256` as well as for our own `Fp` and `Fq`. These are supposed to add no real overhead as most functions are inlined:

```
Proof creation/proof creation (SRS size 2^15)                                                                          
                        time:   [3.3081 s 3.3659 s 3.4263 s]
                        change: [-0.7666% +2.6613% +5.8995%] (p = 0.15 > 0.05)
                        No change in performance detected.
```

(compare this with https://github.com/o1-labs/proof-systems/pull/750)

The benefit is that we can implement ocaml traits on these.

I also implemented serde serialize/deserialize on them, but I didn't manage to remove the serde_as/serde_with stuff elsewhere. I think it's doable though.

### CODE REVIEW

**Comment by mimoo** [comment]
looks like no overhead detected:

```
Proof creation/proof creation (SRS size 2^15)                                                                          
                        time:   [3.3081 s 3.3659 s 3.4263 s]
                        change: [-0.7666% +2.6613% +5.8995%] (p = 0.15 > 0.05)
                        No change in performance detected.
```

(compare with https://github.com/o1-labs/proof-systems/pull/750)

**Review by mrmr1993 — APPROVED** [review]
Really nice!
As a sanity check, it would be good to make a Mina PR and make sure that CI is happy there

**Comment by mimoo** [comment]
playing with it here: https://github.com/MinaProtocol/mina/pull/11906

it seems to compile the stubs. Not sure if mina compiles yet, and haven't touched the wasm part.

File: curves/src/pasta/arkworks/fields.rs [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+                    }
+                }
+
+                //
+                // Helpers
+                //
+
+                #[ocaml_gen::func]
+                #[ocaml::func]
+                pub fn [<$name:snake _size_in_bits>]() -> ocaml::Int {
```
Comment (mrmr1993) [inline curves/src/pasta/arkworks/fields.rs:84]: Per my message OOB: I don't think the bindings for Mina should live here. The behaviour and API are mina-specific, but this library is supposed to be general-purpose.

**Review by mrmr1993 — CHANGES_REQUESTED** [review]
See comment. Can you rework this so that it's only moving the OCaml types into this repo, and leave the rest on the Mina side?

**Comment by mimoo** [comment]
probably we should try this approach first: https://github.com/arkworks-rs/algebra/pull/506

**Comment by mimoo** [comment]
ping @mrmr1993

**Comment by mimoo** [comment]
urg actually I don't think this will solve all of our problems for serde at least... Because things like `G::Scalar` are not bound by `Serialize/Deserialize`. We might have to do the same for `AffineCurve`

**Pull request closed without merging**